src/puppetlabs/puppetdb/query/reports.clj

```diff
@@ -105,6 +86,40 @@
    :logs
    :certname])
 
+(defn logs-query->sql
+  "Converts a vector-structured `query` to a corresponding SQL query which will
+  return nodes matching the `query`."
+  ([version query]
+   (logs-query->sql version query {}))
+  ([version query paging-options]
+   {:pre  [((some-fn nil? sequential?) query)]
+    :post [(map? %)
+           (jdbc/valid-jdbc-query? (:results-query %))
+           (or (not (:count? paging-options))
+               (jdbc/valid-jdbc-query? (:count-query %)))]}
+   (paging/validate-order-by! [:logs] paging-options)
```

Comment:
Huh, add limit & offset to that list of things that do not work for these endpoints yet. Maybe we should just fail on any of the basic paging-options ... makes this all a bit special.